linera-service/src/cli_wrappers/docker.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::util::CommandExt;
+use anyhow::Result;
+use pathdiff::diff_paths;
+use std::path::PathBuf;
+use tokio::process::Command;
+
+pub struct DockerImage {
+    pub name: String,
+}
+
+impl DockerImage {
+    pub fn new(name: String) -> Self {
+        Self { name }
+    }
+
+    pub async fn build(&self, bin_path: &PathBuf, github_root: &PathBuf) -> Result<()> {
+        let bin_path =
+            diff_paths(bin_path, github_root).expect("Getting relative path should not fail");
+        let bin_path = bin_path.to_str().expect("Getting str should not fail");
+
+        let binaries_arg = format!("binaries={bin_path}");
+        let binaries_arg = binaries_arg.as_str();
```

Comment:
Do we need `diff_paths`? It seems that we could just turn `bin_path` into an absolute path (based on the current directory).

nit:  Also, we could probably be a little more concise and inline some of the `let`s (say, the second and the last one). Given that the function returns a `anyhow::Result`, the `.expect(..)`s should also probably be replaced by `.context("while running <function>")?`.

Reply:
(same comments in other files)

Reply:
So we need to execute from the root of the repo, as I'm doing with `current_dir` below, so everything we need is in the Docker build context (it gets everything in the directory you execute it from, including subdirectories, etc). So we need the relative path, because the `Dockerfile` assumes that it'll be a relative path, because it only knows about what's inside the build context. So I think `diff_paths` is needed here. Helm has a similar issue, it needs to be executed from the directory of the `Chart` file, and will only understand relative paths from there.

On the inlining, I initially tried these all in the same line, but the compiler doesn't like it and tells me to put this first part of it in a separate variable :/ Which is why I did it this way. 

TIL about `.context`, thank you! Will do the change

Reply:
> On the inlining, I initially tried these all in the same line, but the compiler doesn't like it and tells me to put this first part of it in a separate variable :/ Which is why I did it this way.

Yes, that's why I suggested to inline the second parts

Reply:
Ahhh I see what you mean now, my bad! Yeah, you're right, that's a lot better. Let me fix it!